system/ui/lib/list_view.py

```diff
@@ -127,34 +156,43 @@ class ListItem:
   rect: "rl.Rectangle" = rl.Rectangle(0, 0, 0, 0)
   callback: Callable | None = None
   action_item: ItemAction | None = None
+  visible: bool | Callable[[], bool] = True
 
   # Cached properties for performance
   _prev_max_width: int = 0
   _wrapped_description: str | None = None
   _prev_description: str | None = None
   _description_height: float = 0
 
-  def actiion(self) -> ItemAction | None:
-    return self.action_item
+  @property
+  def is_visible(self) -> bool:
+    return bool(_resolve_value(self.visible, True))
 
   def get_description(self):
     return _resolve_value(self.description, None)
 
   def get_item_height(self, font: rl.Font, max_width: int) -> float:
+    if not self.is_visible:
+      return 0
+
     current_description = self.get_description()
     if self.description_visible and current_description:
-      if not self._wrapped_description or current_description != self._prev_description or max_width != self._prev_max_width:
+      if (
+        not self._wrapped_description
+        or current_description != self._prev_description
+        or max_width != self._prev_max_width
+      ):
         self._prev_max_width = max_width
         self._prev_description = current_description
 
         wrapped_lines = wrap_text(font, current_description, ITEM_DESC_FONT_SIZE, max_width)
         self._wrapped_description = "\n".join(wrapped_lines)
-        self._description_height = len(wrapped_lines) * ITEM_DESC_FONT_SIZE + 10  # Line height + padding
+        self._description_height = len(wrapped_lines) * ITEM_DESC_FONT_SIZE + 10
       return ITEM_BASE_HEIGHT + self._description_height - (ITEM_BASE_HEIGHT - ITEM_DESC_V_OFFSET) + ITEM_PADDING
     return ITEM_BASE_HEIGHT
 
   def get_content_width(self, total_width: int) -> int:
-    if self.action_item:
+    if self.action_item and self.action_item.get_width() > 0:
```

Comment:
Can we move some of these custom properties into Widget? I want to be able to add any arbitrary Widget into a list view in the future, to allow more flexible design

Reply:
doesn;t have to be in this PR btw

Reply:
hat’s a solid idea! I’ll take some time to refine the design to keep it clean and adaptable for adding any type of Widget. I’ll follow up soon with a proposed solution.